stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/channel/list/adapter/BaseChannelListItemAdapter.kt

```diff
@@ -0,0 +1,37 @@
+package io.getstream.chat.android.ui.channel.list.adapter
+
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.RecyclerView
+import io.getstream.chat.android.client.models.Channel
+import io.getstream.chat.android.ui.channel.list.ChannelListView
+import io.getstream.chat.android.ui.channel.list.ChannelListViewStyle
+import io.getstream.chat.android.ui.channel.list.adapter.diff.ChannelListDiffCallback
+import io.getstream.chat.android.ui.channel.list.adapter.viewholder.BaseChannelListItemViewHolder
+
+public abstract class BaseChannelListItemAdapter :
+    RecyclerView.Adapter<BaseChannelListItemViewHolder>(),
+    ChannelListViewClickDelegate {
+
+    protected open var channels: List<Channel> = emptyList()
+
+    override var userClickListener: ChannelListView.UserClickListener? = null
+
+    override var channelClickListener: ChannelListView.ChannelClickListener? = null
+
+    override var channelLongClickListener: ChannelListView.ChannelClickListener? = null
+
+    public open var style: ChannelListViewStyle? = null
```

Comment:
Why the adapter needs the `style`?
It is not used within the adapter, it should be part of the ViewHolderFactory instead

Reply:
I think so as well, but I wasn't sure how much I could change things.